src/Symfony/Component/Validator/Constraints/NoBannedWordsValidator.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+
+final class NoBannedWordsValidator extends ConstraintValidator
+{
+    public function validate(#[\SensitiveParameter] mixed $value, Constraint $constraint): void
+    {
+        if (!$constraint instanceof NoBannedWords) {
+            throw new UnexpectedTypeException($constraint, NoBannedWords::class);
+        }
+
+        if (null === $value) {
+            return;
+        }
+
+        if (!\is_string($value)) {
+            throw new UnexpectedValueException($value, 'string');
+        }
+
+        $matches = [];
+        if (\count($constraint->dictionary) > 0) {
+            $regex = sprintf(
+                '/%s/i',
+                implode('|', $this->convertToL33t($constraint->dictionary))
```

Comment:
`implode('|', array_map($this->toL33tRegex(...), $constraint->dictionary))` + no need for the method + let's put this sprintf on one line